util/file.py

```diff
@@ -321,6 +321,11 @@ def mkdir_p(dirpath):
         else:
             raise
 
+def touch_p(path):
+    '''Touch file, making parent directories if they don't exist.'''
+    mkdir_p(os.path.dirname(path))
+    open(path, 'a').close()
```

Comment:
For creating files this is fine, but simply opening and closing the file will not update the file times as `touch` would. The function `touch(fname, times=None)` within this same file will do so. I don't see `touch_p` used anywhere else in this PR. Is it needed elsewhere?